impeller/aiks/canvas.cc

```diff
@@ -109,6 +120,11 @@ const Matrix& Canvas::GetCurrentTransformation() const {
   return xformation_stack_.back().xformation;
 }
 
+const Rect Canvas::GetCurrentLocalClipBounds() const {
+  Matrix inverse = xformation_stack_.back().xformation.Invert();
+  return xformation_stack_.back().clip_bounds.TransformBounds(inverse);
+}
```

Comment:
Might be good to add a regular test (doesn't use `OpenPlaygroundHere`) in `aiks_unittests` which uses this method to verify the intersection.

Reply:
Is there a way I can have a test that creates a wide ranging DL, then dispatches it to an impeller Dispatcher with a cull_rect (and one that does that with a sub-DisplayList as well), and then checks that there are only Entities for the ops that should have made it through the culling?

Reply:
I created a new `canvas_unittests` file and wrote a whole suite of tests which found several bugs in Rect::CutOut().

Reply:
With respect to making sure that the Entities and Passes are culled by the cull rect, we already have tests of the DisplayList RTree mechanism that make sure that the culled `Dispatch` variant correctly omits ops in the playback, and with the new tests in `canvas_unittests.cc` that check that Impeller's `Canvas` manages the cull rect according to the desired behavior - is there anything left to test here?

Reply:
I'd say this level of test is beyond the call of action. 😆

Reply:
Well, given that it found errors in an existing geom method, isn't it just "at the call of duty"?

Reply:
We already cover rect methods in geometry_unittests.cc. It sounds like I probably missed a case there.